src/picongpu/include/plugins/common/stringHelpers.hpp

```diff
@@ -50,4 +58,86 @@ namespace picongpu
         return dateString.str();
     }
 
+#if(ENABLE_HDF5 == 1)
+    /** Create c-strings suitable for libSplash
+     *
+     * Convert a std::list of strings to a format that is suitable to
+     * be written into libSplash attributes (\0-concated and padded
+     * array of constant c-strings).
+     * Each string will be padded to the length of the longest string.
+     *
+     * Independent of the padding you chose, the final strings will be '\0'
+     * terminated.
+     */
+    class GetSplashArrayOfString
+    {
+    private:
+        // compare two std::string by their size
+        struct CompStrBySize
+        {
+            bool operator()( std::string i, std::string j)
+            {
+                return i.size() < j.size();
+            }
+        };
+
+    public:
+        // resulting type containing all attributes for a libSplash write call
+        struct Result
+        {
+            splash::ColTypeString colType;
+            std::vector<char*> listIndex;
+            std::vector<char> buffers; // all of same length
+        };
+
+        Result operator()(
+            std::list<std::string> listOfStrings,
+            char padding = '\0'
+        )
+        {
+            // find length of longest string in list
+            CompStrBySize compStrBySize;
+            std::string longestString = *std::max_element(
+                    listOfStrings.begin(),
+                    listOfStrings.end(),
+                    compStrBySize
+            );
+            size_t lenMax = longestString.size();
+
+            // create libSplash data type suitable for longest string
+            Result result;
+            result.colType = splash::ColTypeString(lenMax);
```

Comment:
@ax3l should that not be `splash::ColTypeString(lenMax+1)` since your string is `N+1` bytes long?